python/pyspark/sql/connect/dataframe.py

```diff
@@ -1775,6 +1775,10 @@ def _to_table(self) -> Tuple["pa.Table", Optional[StructType]]:
         assert table is not None
         return (table, schema)
 
+    def _toArrow(self) -> "pa.Table":
```

Comment:
@HyukjinKwon I think it would be fine to make this a regular public method instead of the `_toArrow` maybe change it to `toArrowTable` to give a better indication of the return type.

Reply:
Yeah, the current state looks a bit weird. We should either explicitly add an developer API, or just don't. My only question is that we can work around via `SparkSession.client.to_table()`, and we're going forward to Spark Connect default. Considering this is a developer API, I wonder if we should add this into DataFrame at this moment.

Reply:
I updated the PR to make the method public and rename it `toArrowTable`.

I will also push a commit here that adds some relevant content to the [Apache Arrow in PySpark](https://spark.apache.org/docs/latest/api/python/user_guide/sql/arrow_pandas.html) page.

Reply:
I expect that this will be used by some end users, not only by developers. In recent years, PyArrow has become a very widely used library (>100M downloads/month on PyPI, ~50th most popular library) and we see a lot of non-developers using it.